recipes/openscenegraph/all/conanfile.py

```diff
@@ -163,75 +153,85 @@ def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename("OpenSceneGraph-OpenSceneGraph-" + self.version, self._source_subfolder)
 
-    def build(self):
-        cmake = CMake(self)
+        for patch in self.conan_data["patches"].get(self.version, []):
+            tools.patch(**patch)
 
-        cmake.definitions["DYNAMIC_OPENSCENEGRAPH"] = self.options.shared
-        cmake.definitions["DYNAMIC_OPENTHREADS"] = self.options.shared
+        for package in ("Fontconfig", "Freetype", "GDAL", "GIFLIB", "GTA", "Jasper", "OpenEXR"):
+            # Prefer conan's find package scripts over osg's
+            os.unlink(os.path.join(self._source_subfolder, "CMakeModules", "Find{}.cmake".format(package)))
 
-        cmake.definitions["BUILD_OSG_APPLICATIONS"] = self.options.build_applications
-        cmake.definitions["BUILD_OSG_EXAMPLES"] = self.options.build_examples
+    def toolchain(self):
+        toolchain = CMakeToolchain(self)
```

Comment:
IIRC this requires a very recent CMake version. I think the regular CMake build helper with our common CMake wrapper is a better choice

Reply:
CMake itself has had support for toolchain files for a long time (at least since 3.0). Are you referring to the contents of conan's toolchain file?

Reply:
This is too early and experimental, please don't merge it.

Toolchains will change in next 1.30, this variable will be called ``preprocessor_definitions`` to make clear the intent.

Reply:
@memsharded I think `definitions` maps roughly to `variables` in 1.30, not `preprocessor_definitions`. There doesn't seem to be an exact equivalent looking at the [commit](https://github.com/conan-io/conan/commit/a77b97817a73fbdd2fbeb5f7038ad400aafd1803), as `definitions` set CMake cache variables (i.e. as if they had been passed via the command line), where as `variables` just sets normal CMake variables.

Nevertheless, I take your point. I'll see if things work using the old `cmake` generator.